e2e/e2e-windows-scenario.sh

```diff
@@ -0,0 +1,179 @@
+    --argjson JsonForIdentity "$WINDOWS_IDENTITY" \
+    --argjson JsonForNetwork "$NETWORK_PROPERTIES" \
+    --arg ValueForAdminPassword "$WINDOWS_PASSWORD" \
+    --arg ValueForCustomData "$CUSTOM_DATA" \
+    --arg ValueForCSECmd "$CSE_CMD" \
+    --arg ValueForVMSS "$DEPLOYMENT_VMSS_NAME" \
+    '.parameters += $JsonForVnet | .parameters += $JsonForLB | .resources[0] += $JsonForIdentity | .resources[0].properties.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0] += $JsonForNetwork | .resources[0].properties.virtualMachineProfile.osProfile.adminPassword=$ValueForAdminPassword | .resources[0].properties.virtualMachineProfile.osProfile.customData=$ValueForCustomData | .resources[0].properties.virtualMachineProfile.extensionProfile.extensions[0].properties.settings.commandToExecute=$ValueForCSECmd | .parameters.virtualMachineScaleSets_akswin30_name.defaultValue=$ValueForVMSS' \
+    windows_vmss_template.json > deployment.json
+
+set +e
+az deployment group create --resource-group $MC_RESOURCE_GROUP_NAME \
+         --template-file deployment.json
+retval=$?
+set -e
+
+cat $SCENARIO_NAME-vmss.json
+
+VMSS_INSTANCE_NAME=$(az vmss list-instances \
+                    -n ${DEPLOYMENT_VMSS_NAME} \
+                    -g $MC_RESOURCE_GROUP_NAME \
+                    -ojson | \
+                    jq -r '.[].osProfile.computerName')
+export DEPLOYMENT_VMSS_NAME
+export VMSS_INSTANCE_NAME
+
+VMSS_INSTANCE_ID=$(az vmss list-instances \
+                    -n ${DEPLOYMENT_VMSS_NAME} \
+                    -g $MC_RESOURCE_GROUP_NAME \
+                    -ojson | \
+                    jq -r '.[].instanceId'
+                )
+export VMSS_INSTANCE_ID
+
+# FAILED=0
+# # Check if the node joined the cluster
+# if [[ "$retval" != "0" ]]; then
+#     err "cse failed to apply"
+#     debug
+#     tail -n 50 $SCENARIO_NAME-logs/cluster-provision.log || true
+#     exit 1
+# fi
+
+# Sleep to let the automatic upgrade of the VM finish
+waitForNodeStartTime=$(date +%s)
+for i in $(seq 1 10); do
+    set +e
+    kubectl get nodes | grep $VMSS_INSTANCE_NAME
+    retval=$?
+    # pipefail interferes with conditional.
+    # shellcheck disable=SC2143
+    if [ -z "$(kubectl get nodes | grep $VMSS_INSTANCE_NAME | grep -v "NotReady")" ]; then
+        log "retrying attempt $i"
+        sleep 10
+        continue
+    fi
+    break;
+done
+waitForNodeEndTime=$(date +%s)
+log "Waited $((waitForNodeEndTime-waitForNodeStartTime)) seconds for node to join"
+
+FAILED=0
+# Check if the node joined the cluster
+if [ "$retval" -eq 0 ]; then
+    ok "Test succeeded, node joined the cluster"
+    kubectl get nodes -o wide | grep $VMSS_INSTANCE_NAME
+else
+    err "Node did not join cluster"
+    FAILED=1
+fi
+
+# Run a windows servercore pod on the node to check if pod runs
+POD_NAME=$(mktemp -u podName-XXXXXXX | tr '[:upper:]' '[:lower:]')
+export POD_NAME
+envsubst < pod-windows-template.yaml > pod-windows.yaml
+sleep 5
```

Comment:
curious: why do we need to sleep 5 seconds? Does envsubst need so long to write the file to the disk?

Reply:
Actually I refer to the code in linux pipeline and I guess the time is used to avoid that file is not ready?